### PR TITLE
Is575/logging stack

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

Creates a logging stack made of graylog-server, elasticsearch, mongodb and logspout.

- [x] added graylog stack
- [x] updated makefile of deployer to create the swarm only if necessary

## Related issue number

links osparc-simcore#575

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] **Runs in the swarm**
- [ ] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: services/graylog/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+```json
+    "log-driver": "gelf",
+    "log-opts": {
+      "gelf-address": "udp://127.0.0.1:12201"
+    }
+```
+
+Restart the docker daemon after the modifications.
+
+**Note:** This has the downside that all commands relying on "docker logs" will then fail.
```
Comment (mguidon) [inline services/graylog/README.md]: Did you not say there is a workaround for that problem?

File: services/deployment-agent/Makefile [hunk 2]
```diff
@@ -1,98 +1,73 @@
...
+	docker-compose -f docker-compose.yml -f docker-compose.devel.yaml build --parallel
 
-.PHONY: up up-devel down
-# target: up, up-devel, down: – Starts/Stops services.
-up-devel: .env deployment_config.yaml
-	${DOCKER} swarm init
-	${DOCKER_COMPOSE} -f docker-compose.yml -f docker-compose.devel.yaml config > $(TEMPCOMPOSE).tmp-compose.yml
-	${DOCKER} stack deploy -c $(TEMPCOMPOSE).tmp-compose.yml portainer
+.PHONY: up up-devel
+up-devel: init .env deployment_config.yaml ## target: up, up-devel, down: – Starts/Stops services.
```
Comment (pcrespov) [inline services/deployment-agent/Makefile]: Minor: no need to write "target" in all the help entries to get it identified as help. Also, every target can has its own input

File: services/graylog/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      # Password: admin
+      - GRAYLOG_ROOT_PASSWORD_SHA2=8c6976e5b5410415bde908bd4dee15dfb167a9c873fc4bb8a81f6f2ab448a918
+      - GRAYLOG_HTTP_EXTERNAL_URI=http://127.0.0.1:19000/
+    depends_on:
+      - mongodb
+      - elasticsearch
+    ports:
+      # Graylog web interface and REST API
+      - 19000:9000
+      # Syslog TCP
```
Comment (pcrespov) [inline services/graylog/docker-compose.yml]: Minor: should we open all these ports, even if not used?

File: services/graylog/README.md (hunk 1 above)
Comment (pcrespov) [reply services/graylog/README.md]: Yes, I also thought that Logspout-GELF would solve this issue

Comment (sanderegg) [reply services/graylog/README.md]: It solves the issue yes. This is here for mere information. I will precise this point.

File: services/graylog/docker-compose.yml (hunk 3 above)
Comment (sanderegg) [reply services/graylog/docker-compose.yml]: syslog will be used for the machine logs I think.
Also as everything is anyway behind nginx, that should not harm I think

File: services/deployment-agent/Makefile (hunk 2 above)
Comment (sanderegg) [reply services/deployment-agent/Makefile]: done

**Pull request merged**